config/ldp/authorization/readers/acl-plus.json

```
@@ -0,0 +1,41 @@
+{
+  "@context": "https://linkedsoftwaredependencies.org/bundles/npm/@solid/community-server/^5.0.0/components/context.jsonld",
+  "import": [
+    "css:config/ldp/authorization/readers/access-checkers/agent.json",
+    "css:config/ldp/authorization/readers/access-checkers/client.json",
+    "css:config/ldp/authorization/readers/access-checkers/issuer.json"
+  ],
+  "@graph": [
+    {
+      "comment": "Adds parent container checks needed for create/delete permissions.",
+      "@id": "urn:solid-server:default:WrappedWebAclReader",
+      "@type": "ParentContainerReader",
+      "identifierStrategy": { "@id": "urn:solid-server:default:IdentifierStrategy" },
+      "reader": { "@id": "urn:solid-server:default:WebAclAuxiliaryReader" }
+    },
+    {
+      "comment": "Reinterprets Control permissions as Read/Write on the ACL document.",
+      "@id": "urn:solid-server:default:WebAclAuxiliaryReader",
+      "@type": "AuthAuxiliaryReader",
+      "authStrategy": { "@id": "urn:solid-server:default:AclStrategy" },
+      "reader": { "@id": "urn:solid-server:default:WebAclReader" }
+    },
+    {
+      "comment": "Reads out permissions from an ACL document for subject resources.",
+      "@id": "urn:solid-server:default:WebAclReader",
+      "@type": "WebAclReader",
+      "aclStrategy": { "@id": "urn:solid-server:default:AclStrategy" },
+      "resourceSet": { "@id": "urn:solid-server:default:CachedResourceSet" },
+      "aclStore": { "@id": "urn:solid-server:default:ResourceStore" },
+      "identifierStrategy": { "@id": "urn:solid-server:default:IdentifierStrategy" },
+      "accessChecker": {
+        "@type": "BooleanAndHandler",
+        "handlers": [
+          { "@id": "urn:solid-server:default:AgentAccessCheckers" },
+          { "@id": "urn:solid-server:default:ClientAccessCheckers" },
+          { "@id": "urn:solid-server:default:IssuerAccessCheckers" }
+        ]
```

Does that mean that every ACL authorization block now needs to define permissions for agents, clients and issuers, and if it excludes one of those the authorization block will automatically reject all requests?

In general we also use singular names for identifiers as it shouldn't matter if it's a combination of entries or not.

```suggestion
        "@type": "BooleanAndHandler",
        "handlers": [
          { "@id": "urn:solid-server:default:AgentAccessChecker" },
          { "@id": "urn:solid-server:default:ClientAccessChecker" },
          { "@id": "urn:solid-server:default:IssuerAccessChecker" }
        ]
```

---

Yes, but this was just an example. In practice, I think this will turn out to be required for agents and clients, and optional for issuers. We could also overcome this practical "burden" by having the ACL template initially include the classes for any client/issuer.